test/dict.jl

```diff
@@ -151,7 +151,7 @@ end
 let
     d = Dict{String, Vector{Int}}()
     d["a"] = [1, 2]
-    @test_throws MethodError d["b"] = 1
```

Comment:
This is comes from the fact that scalars are iterable ATM. We could get rid of these two behaviours at the same time.